File: 226-invert-binary-tree/226-invert-binary-tree.scala
/**
 * Definition for a binary tree node.
 * class TreeNode(_value: Int = 0, _left: TreeNode = null, _right: TreeNode = null) {
 *   var value: Int = _value
 *   var left: TreeNode = _left
 *   var right: TreeNode = _right
 * }
 */
object Solution {
    def invertTree(root: TreeNode): TreeNode = {
        helper(root)
        root
    }
    
    def helper(root: TreeNode): Unit = {
        if (root != null) {
            helper(root.left)
            helper(root.right)
            val left = root.left
            root.left = root.right
            root.right = left
        }
    }
}

File: 226-invert-binary-tree/226-invert-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public TreeNode invertTree(TreeNode root) {
        Queue<TreeNode> queue = new LinkedList<>();
        
        if (root != null) queue.offer(root);
        
        while(!queue.isEmpty()) {
            TreeNode node = queue.poll();
            TreeNode left = node.left;
            TreeNode right = node.right;
            
            node.left = right;
            node.right = left;
            
            if (left != null) queue.offer(left);
            if (right != null) queue.offer(right);
        }
        
        return root;
    }
}